### ISSUE TITLE
visibility_detector Dependency conflicts [BUG]

### ISSUE THREAD

**Issue by amalshyjo** [jhomlala/betterplayer#1283]
**History check**
Please confirm that you've checked issues history and you didn't find anything which may solve your issue.

**Describe the bug**
visibility_detector package used by betterplayer is 0.3.3 but most packages have updated to 0.4.0+2 causing dependency conflicts with multiple packages like UX Cam, etc...

**To Reproduce**
1. Go to you flutter project and increase visibility_detector version from 0.3.3 to 0.4.0+2 
2. Run flutter pub get
3. The package getter will fail with Dependency conflicts issue where there are multiple apps with support for 0.4.0+2 causing the packages unusable 
4. 
`ecause better_player >=0.0.62 <0.0.68 depends on visibility_detector ^0.2.0-nullsafety.1 and better_player >=0.0.68 <0.0.80 depends on visibility_detector ^0.2.0, better_player >=0.0.62 <0.0.80 requires visibility_detector ^0.2.0-nullsafety.1.
And because better_player >=0.0.80 <0.0.83 depends on visibility_detector ^0.2.2 and better_player >=0.0.83 depends on visibility_detector ^0.3.3, better_player >=0.0.62 requires visibility_detector ^0.2.0-nullsafety.1 or ^0.3.3.`

**Example code*
` better_player: ^0.0.66`
` visibility_detector: ^0.4.0+2`

**Expected behavior**
Need to update betterplayer package to use  visibility_detector: ^0.4.0+2

**Screenshots**

**Flutter doctor**
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.7.8, on macOS 13.0.1 22A400 darwin-arm64, locale en-IN)
[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.3)
[✓] VS Code (version 1.85.1)
[✓] Connected device (2 available)
    ! Error: Browsing for AKHILA’s iPhone (CoreDevice)  on the local area network. Ensure that the device is booted, unlocked, and discoverable via Bluetooth and WiFi, or attached with a cable. (code -27)
[✓] HTTP Host Availability

• No issues found!

**Better Player version**
- Version: <0.0.83

**Smartphone (please complete the following information):**
 - Device: any device
 - OS: any OS

**Additional context**

**Comment by amalshyjo** [comment]
fixed #1284

**Comment by fabioselau077** [comment]
same with flutter_secure_storage package...

**Comment by sajedulsakib001** [comment]
As you said, if I just change the version in betterplayer/pubspec.yaml, will it fix the issue?

**Comment by fabioselau077** [comment]
> As you said, if I just change the version in betterplayer/pubspec.yaml, will it fix the issue?

try [better_player_plus](https://pub.dev/packages/better_player_plus)

**Comment by amalshyjo** [comment]
> As you said, if I just change the version in betterplayer/pubspec.yaml, will it fix the issue?

Yes unless there is a syntax change/ breaking change in the newer version just updating the pub version should do it. But the merges haven't been published yet

**Comment by jhomlala** [comment]
Please use latest BP version.
